### PR TITLE
Minor proof fixes

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview
Minor changes in proofs.

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding-common/src/test/java/com/exonum/binding/common/proofs/map/UncheckedFlatMapProofTest.java [hunk 1]
```diff
@@ -94,10 +94,10 @@ void mapProofWithSeveralLeafsShouldBeCorrect() {
         new MapEntry(fourthKey, THIRD_VALUE)
     );
 
-    Set<MapEntry> actualCheckedEntriesList = checkedMapProof.getEntries();
+    Set<MapEntry> actualCheckedEntriesSet = checkedMapProof.getEntries();
```
Comment (dmitry-timofeev) [inline exonum-java-binding-common/src/test/java/com/exonum/binding/common/proofs/map/UncheckedFlatMapProofTest.java]: Better remove the type from the name: `actualPresentEntries`?

File: exonum-java-binding-common/src/test/java/com/exonum/binding/common/proofs/map/UncheckedFlatMapProofTest.java [hunk 2]
```diff
@@ -94,10 +94,10 @@ void mapProofWithSeveralLeafsShouldBeCorrect() {
...
 
     assertThat(
-        actualCheckedEntriesList,
+        actualCheckedEntriesSet,
```
Comment (dmitry-timofeev) [inline exonum-java-binding-common/src/test/java/com/exonum/binding/common/proofs/map/UncheckedFlatMapProofTest.java]: Why are `expectedEntriesList` the same as leaves? Is it intended?

(I'd also drop the suffix specifying the type)

File: exonum-java-binding-common/src/main/java/com/exonum/binding/common/proofs/map/CheckedMapProof.java [hunk 3]
```diff
@@ -33,7 +33,7 @@
  * // Convert to checked
  * CheckedMapProof checkedProof = proof.check();
  * // Check the root hash
- * if (checkedProof.compareWithRootHash(expectedRootHash)) {
+ * if (checkedProof.isValid() && checkedProof.compareWithRootHash(expectedRootHash)) {
```
Comment (dmitry-timofeev) [inline exonum-java-binding-common/src/main/java/com/exonum/binding/common/proofs/map/CheckedMapProof.java]: Haven't we removed `compareWithRootHash`?

Comment (MakarovS) [reply exonum-java-binding-common/src/main/java/com/exonum/binding/common/proofs/map/CheckedMapProof.java]: We did in `CheckedListProof`, not in `CheckedMapProof`. I can do it in this PR, if we want to remove it?

Comment (dmitry-timofeev) [reply exonum-java-binding-common/src/main/java/com/exonum/binding/common/proofs/map/CheckedMapProof.java]: :+1:

**Pull request merged**